### PR TITLE
Rxpointers

### PR DESCRIPTION
**Pull request by thattil**
- remove pointers being sent as arguments to Listener, DataProcessor and DataStreamer classes
- remove redundant class members from Implementation class that are already there in generalData
- reduce arguments and data members created in HDF5Data and BinaryData class to create file name prefix at Implementation before passing to DataProcessor (filepath, filename, detIndex, numUnitsPerDetector). Master file name kept as they are.

### CODE REVIEW

File: slsReceiverSoftware/src/BinaryDataFile.h [hunk 1]
```diff
@@ -14,12 +14,10 @@ class BinaryDataFile : private virtual slsDetectorDefs, public File {
...
     fileFormat GetFileFormat() const override;
     void CloseFile() override;
-    void CreateFirstBinaryDataFile(const std::string fPath,
-                                   const std::string fNamePrefix,
+    void CreateFirstBinaryDataFile(const std::string fNamePrefix,
                                    const uint64_t fIndex,
                                    const bool ovEnable,
-                                   const bool sMode, const int modulePos,
-                                   const int nUnitsPerReadout,
+                                   const bool sMode,
```
Comment (erikfrojdh) [inline slsReceiverSoftware/src/BinaryDataFile.h:20]: Fun fact: when passing by value (and not reference) the const on parameters in the function declarations doesn't do anything. Only the implementation matters. In principle we could remove the const from const bool in .h and only keep it in the .cpp file.

**Review by erikfrojdh — COMMENTED** [review]
Maybe fix the const& for the strings and then merge?

File: slsReceiverSoftware/src/BinaryDataFile.cpp [hunk 2]
```diff
@@ -20,22 +20,17 @@ void BinaryDataFile::CloseFile() {
 }
 
 void BinaryDataFile::CreateFirstBinaryDataFile(
-    const std::string fPath, const std::string fNamePrefix,
+    const std::string fNamePrefix,
```
[the remark this answers is not in the document]
Comment (thattil) [reply slsReceiverSoftware/src/BinaryDataFile.cpp:23]: done

File: slsReceiverSoftware/src/BinaryDataFile.h [hunk 3]
```diff
@@ -14,12 +14,10 @@ class BinaryDataFile : private virtual slsDetectorDefs, public File {
 
     fileFormat GetFileFormat() const override;
     void CloseFile() override;
-    void CreateFirstBinaryDataFile(const std::string fPath,
-                                   const std::string fNamePrefix,
+    void CreateFirstBinaryDataFile(const std::string fNamePrefix,
```
[the remark this answers is not in the document]
Comment (thattil) [reply slsReceiverSoftware/src/BinaryDataFile.h:17]: done

**Pull request merged**